integration_test.go

```diff
@@ -40,19 +40,52 @@ func TestIntegrationAPI(t *testing.T) {
 
 		name := keys[i]
 		tData := circuits.Circuits[name]
-		assert.Run(func(assert *test.Assert) {
-			for i := range tData.ValidAssignments {
-				assert.Run(func(assert *test.Assert) {
-					assert.ProverSucceeded(tData.Circuit, tData.ValidAssignments[i], test.WithProverOpts(backend.WithHints(tData.HintFunctions...)), test.WithCurves(tData.Curves[0], tData.Curves[1:]...))
-				}, fmt.Sprintf("valid-%d", i))
-			}
-
-			for i := range tData.InvalidAssignments {
-				assert.Run(func(assert *test.Assert) {
-					assert.ProverFailed(tData.Circuit, tData.InvalidAssignments[i], test.WithProverOpts(backend.WithHints(tData.HintFunctions...)), test.WithCurves(tData.Curves[0], tData.Curves[1:]...))
-				}, fmt.Sprintf("invalid-%d", i))
-			}
-		}, name)
+
+		// Plonk + FRI is tested only for Mul circuit (otherwise it slows everything down a lot...)
```

Comment:
is it still needed? If so, factorize code;
if name == "mul" --> backends := [plonk, plonk-fri, groth16]
"test.WithBackends(backends...)"